### PR TITLE
Add new CLI auth flow w/ tests and docs

### LINKED ISSUE #1417: More Comprehensive Auth CLI commands
## Use Case
Currently `prefect auth` only contains a `login` command and we need to support all the new hooks in #1410 

## Solution
`prefect auth logout` (logout tenant, delete token) w/ confirm prompt
`prefect auth list` (list available tenants)
`prefect auth switch` (switch tenant)
`prefect auth create-token` (create token scoped to a role)
`prefect auth revoke-token` (revoke an API token)

login w/ user token -> verify user

show list of tenants -> choose tenant

## Alternatives
N/A

### PR DESCRIPTION
**Pull request by joshmeek**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Closes #1417 
Adds new CLI commands for working with Prefect Cloud auth
Related to changes in #1423

e.g.
```
        $ prefect auth login --token MY_TOKEN
        Log in successful

        $ prefect auth list-tenants
        NAME                        SLUG                        ID
        Test Person                 test-person                 816sghf2-4d51-4338-a333-1771gns7614d
        test@prefect.io's Account   test-prefect-io-s-account   1971hs9f-e8ha-4a33-8c33-64512gds86g1 

        $ prefect auth switch-tenants --slug test-person
        Tenant switched

        $ prefect auth list-tenants
        NAME                        SLUG                        ID
        Test Person                 test-person                 816sghf2-4d51-4338-a333-1771gns7614d    *
        test@prefect.io's Account   test-prefect-io-s-account   1971hs9f-e8ha-4a33-8c33-64512gds86g1 

        $ prefect auth logout
        Logged out from tenant 

        $ prefect auth list-tenants
        NAME                        SLUG                        ID
        Test Person                 test-person                 816sghf2-4d51-4338-a333-1771gns7614d
        test@prefect.io's Account   test-prefect-io-s-account   1971hs9f-e8ha-4a33-8c33-64512gds86g1 

```

Depends on #1426 

## Why is this PR important?
Makes auth really easy to work with in the CLI

### CODE REVIEW

File: docs/cloud/cloud_concepts/cli.md [hunk 1]
```diff
@@ -373,6 +373,60 @@ $ prefect auth login --token BAD_TOKEN
 Error attempting to use Prefect API token BAD_TOKEN
 ```
 
+#### auth logout
+
+Running `prefect auth logout` will log you out of your active tenant if one if currently logged in to.
```
Comment (cicdw) [inline docs/cloud/cloud_concepts/cli.md]:
```suggestion
Running `prefect auth logout` will log you out of your active tenant (if you are logged in)
```

File: docs/cloud/cloud_concepts/cli.md [hunk 2]
```diff
@@ -373,6 +373,60 @@ $ prefect auth login --token BAD_TOKEN
...
+```
+$ prefect auth list-tenants
+NAME                        SLUG                        ID
+Test Person                 test-person                 816sghf2-4d51-4338-a333-1771gns7614d
+test@prefect.io's Account   test-prefect-io-s-account   1971hs9f-e8ha-4a33-8c33-64512gds86g1  *
+```
+
+#### auth switch-tenants
+
+Running `prefect auth switch-tenants --id TENANT_ID --slug TENANT_SLUG` will switch your active tenants. Either the tenant ID or the tenant slug need to be provided.
```
Comment (cicdw) [inline docs/cloud/cloud_concepts/cli.md]:
```suggestion
Running `prefect auth switch-tenants --id TENANT_ID --slug TENANT_SLUG` will switch your active tenants. Either the tenant ID or the tenant slug needs to be provided.
```

File: docs/cloud/cloud_concepts/cli.md [hunk 3]
```diff
@@ -373,6 +373,60 @@ $ prefect auth login --token BAD_TOKEN
...
+
+If you are unable to switch tenants for various reasons (bad id, bad slug, not providing either) then you should see:
+```
+$ prefect auth switch-tenants --slug test-person
+Unable to switch tenant
+```
+
+#### auth create-token
+
+Running `prefect auth create-token --name MY_TOKEN --role RUNNER` will generate a Prefect Cloud API token and output it to stdout. For more information on API tokens visit go [here](./api.html).
```
Comment (cicdw) [inline docs/cloud/cloud_concepts/cli.md]:
```suggestion
Running `prefect auth create-token --name MY_TOKEN --role RUNNER` will generate a Prefect Cloud API token and output it to stdout. For more information on API tokens go [here](./api.html).
```

**Comment by joshmeek** [comment]
Yeah @cicdw I already love it because it removes the need for multiple manually changing tenant-scoped tokens in my `config.toml`

**Comment by jlowin** [comment]
Quick hits:

- login should automatically use a default membership (if available) to switch tenants off the bat
- disable all auth commands if a token is found in config (because it'll be given preference)
- commands for listing tokens (name, id)
- commands for revoking tokens (by id)

**Comment by joshmeek** [comment]
@jlowin @cicdw Updated as per comments

**Pull request merged**